sql/interface.go

```diff
@@ -6,13 +6,16 @@ import "go.uber.org/zap"
 
 // Executor is an interface for executing raw statement.
 type Executor interface {
+	// Exec executes a statement.
 	Exec(string, Encoder, Decoder) (int, error)
 }
 
 // Migration is interface for migrations provider.
 type Migration interface {
+	// Apply applies the migration.
 	Apply(db Executor, logger *zap.Logger) error
-	Rollback() error
```

Comment:
`Rollback` is actually a quite important feature for some coded migrations. We had a coded migration before that moved data from files on the disk into the database. If something went wrong I want to roll back the renaming of the files or the migration can never be applied again without manual intervention by the user.

Reply:
Reverted this part of the PR